forms/core/CampaignMonitor.php

```diff
@@ -37,29 +37,41 @@ public function get_last_error(){
      */
     public function refresh_token($auth = array())
     {
-	   return $this->api->refreshToken();
+        $refreshToken = empty($auth) ? '' : $auth['refresh_token'];
+        return json_decode($this->api->refreshToken($refreshToken));
     }
     /**
      * @param array $auth override the class authentication credentials
      * @return mixed|null list of clients
      */
     public function get_clients($credentials = array())
     {
+	    if (empty($credentials)) {
+            return (object) ['error' => true];              
+        }
 
-	    if ( ! empty( $credentials ) ) {
-		    $auth = new Authorization();
-		    $auth->setAccessToken($credentials['access_token']);
-		    $auth->setRefreshToken($credentials['refresh_token']);
-		    $auth->setType(Authorization::OAUTH);
+        $auth = new Authorization();
+        $auth->setAccessToken($credentials['access_token']);
+        $auth->setRefreshToken($credentials['refresh_token']);
+        $auth->setType(Authorization::OAUTH);
 
-		    $this->api->setAuthorization( $auth );
-	    }
+        $this->api->setAuthorization( $auth );

```

Comment:
Since you have created a new method update_tokens below, would you be able to use that function instead?